isic-archive-tasks/isic_archive_tasks/__init__.py

```diff
@@ -0,0 +1,103 @@
+from celery import Celery
+from celery.utils.log import get_task_logger
+import jsonpickle
+from kombu.serialization import register
+import sentry_sdk
+
+from girder.constants import TokenScope
+from girder.models.token import Token
+from girder.models.user import User
+from girder.utility import mail_utils
+
+app = Celery()
+
+task_logger = get_task_logger(__name__)
+sentry_sdk.init()
+
+
+@app.on_after_configure.connect
+def setup_periodic_tasks(sender, **kwargs):
+    sender.add_periodic_task(30, sendIngestionNotifications.s(),
+                             name='Send any necessary notifications for ingested batches.')
+
+
+@app.on_after_configure.connect
+def fix_girder_imports(sender, **kwargs):
+    from girder.utility.server import configureServer
+    configureServer(curConfig={
+        'server': {
+            'mode': 'production'
+        },
+        'cherrypy_server': False
+    })
+
+
+class CredentialedGirderTask(app.Task):
+    serializer = 'jsonpickle'  # Supports objectIds
+
+    def __call__(self, *args, **kwargs):
+        from girder.plugins.isic_archive.provision_utility import getAdminUser
+        self.token = Token().createToken(user=getAdminUser(), days=1,
+                                         scope=[TokenScope.DATA_READ, TokenScope.DATA_WRITE])
+        self.token = str(self.token['_id'])
+
+        super(CredentialedGirderTask, self).__call__(*args, **kwargs)
+
+
+register('jsonpickle', jsonpickle.encode, jsonpickle.decode, content_type='application/json',
+         content_encoding='utf-8')
+
+
+@app.task()
+def sendIngestionNotifications():
+    from girder.plugins.isic_archive.models.batch import Batch
+    from girder.plugins.isic_archive.models.image import Image
+    for batch in Batch().find({'ingestStatus': 'extracted'}):
+        if Batch().hasImagesPendingIngest(batch):
+            return
+        else:
+            failedImages = list(Image().find({'$and': [
+                {'meta.batchId': batch['_id']},
+                {'$or': [
+                    {'ingestionState.large-image': False},
+                    {'ingestionState.superpixel-mask': False}]}]}))
+
```

Comment:
Can we run a second query, looking for images with `'readable': False`? We should add this to the email as a list of "skipped files".